contrib/drivers/filesharedrivers/nfs/nfs.go

```diff
@@ -163,28 +165,59 @@ func (d *Driver) CreateFileShare(opt *pb.CreateFileShareOpts) (*model.FileShareS
 		return nil, err
 	}
 
-	if err := d.cli.CreateVolume(name, vg, opt.GetSize()); err != nil {
-		return nil, err
-	}
-	// remove created volume if got error
-	defer func() {
-		// using return value as the error flag
-		if fshare == nil {
-			if err := d.cli.Delete(name, vg); err != nil {
-				log.Error("failed to remove volume fileshare:", err)
-			}
+	if opt.SnapshotId != "" {
```

Comment:
not "" means always valid?

Reply:
It means, we have given some snapshotId through parameter. If not given then always valid and create freshfileshare.